mne/io/brainvision/tests/test_brainvision.py

```diff
@@ -182,7 +204,7 @@ def test_brainvision_data_highpass_filters():
     # Homogeneous highpass in seconds (default measurement unit)
     raw = _test_raw_reader(
         read_raw_brainvision, vhdr_fname=vhdr_highpass_path,
-        montage=montage, eog=eog)
```

Comment:
@massich why do you need to change all these montage param to None?

Reply:
Indeed this change seems strange. Maybe instead have a `pytest.parametrize` over `None` and the actual `montage`?

Reply:
it is true that all those calls were done with a montage. But it was never tested that the montage was giving what was expected. In other words, if I had modified the positions or simply passed a random montage no test was breaking.

So I set them all to `None` (our end goal is to remove the parameter at the end) and I added a test that was exercising the `set_montage`. It is true that I could add a second test that `montage` can be passed to `read_raw_brainvision` and that behaves as expected.

Reply:
> In other words, if I had modified the positions or simply passed a random montage no test was breaking.

This to me just means that the test should be improved. But it was at least a smoke test. The change even gets rid of this smoke testing...

> our end goal is to remove the parameter at the end

But we can do this by doing `raw.set_montage(montage)` instead of `raw = read_raw_x(..., montage=montage)` inside of `_test_raw_reader`. So I don't see why we should remove it, having it is forward compatible with our plan.